ouroboros-network/src/Ouroboros/Network/BlockFetch/State.hs

```diff
@@ -258,7 +264,8 @@ data FetchStateFingerprint peer header block =
        !(Maybe (Point block))
        !(Map peer (Point header))
        !(Map peer (PeerFetchStatus header))
-  deriving Eq
+
+deriving instance (Ord peer, Eq header, Eq (HeaderHash header), Eq (HeaderHash block)) => Eq (FetchStateFingerprint peer header block)
```

Comment:
This seems quite suspicious. Why are we now comparing headers? Surely we only need to compare points.